ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-with-predicates.json

```diff
@@ -224,6 +224,292 @@
           {"row":{"columns":["11", "v11a", "v11b"]}}
         ]}
       ]
+    },
+    {
+      "name": "schema registry table-test one partition json",
+      "statements": [
+        "CREATE TABLE INPUT (ID INT PRIMARY KEY, V0 STRING PRIMARY KEY, V1 STRING) WITH (kafka_topic='test_topic', format='JSON', PARTITIONS=1);",
+        "CREATE TABLE MATVIEW AS SELECT * FROM INPUT;",
+        "SELECT * FROM MATVIEW WHERE ID=10 and V0='v10a';"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  true
+      },
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12345, "key": {"ID": "10", "v0": "v10a"}, "value": {"v1": "v10b"}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`ID` INTEGER KEY, `V0` STRING KEY, `V1` STRING"}},
+          {"row":{"columns":[10, "v10a", "v10b"]}}
+        ]}
+      ]
+    },
+    {
+      "name": "schema registry table-test one partition avro",
+      "statements": [
+        "CREATE TABLE INPUT (ID INT PRIMARY KEY, V0 STRING PRIMARY KEY, V1 STRING) WITH (kafka_topic='test_topic', format='AVRO', PARTITIONS=1);",
+        "CREATE TABLE MATVIEW AS SELECT * FROM INPUT;",
```

Comment:
can we add a test for what happens when we create a table with a different format than the original?

e.g.
```
CREATE TABLE MATVIEW WITH (FORMAT='PROTOBUF') AS SELECT * FROM INPUT;
```